api/rest/client/client.go

```diff
@@ -91,6 +93,9 @@ type Client struct {
 // NewClient initializes a client given a Config.
 func NewClient(cfg *Config) (*Client, error) {
```

Comment:
Instead of adding a `context.Context` to the `Config` struct, I suggest providing it to the `Client` via a parameter here. Putting it in `Config` is a round-about way of setting a private field and `Context` is out of place with the other fields of `Config`.